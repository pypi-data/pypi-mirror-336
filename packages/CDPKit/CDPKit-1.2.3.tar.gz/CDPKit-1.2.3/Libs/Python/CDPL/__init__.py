##
# __init__.py  
#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
##


from __future__ import absolute_import

import os
import sys
import platform

if sys.version_info[1] >= 8 and platform.system() == 'Windows':
    if 'site-packages' in __file__:
        os.add_dll_directory(os.path.dirname(__file__))
    else:
        os.add_dll_directory(os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(__file__))), '@CDPKIT_EXECUTABLE_INSTALL_DIR@'))

from ._cdpl import *

__version__ = VERSION_STRING
