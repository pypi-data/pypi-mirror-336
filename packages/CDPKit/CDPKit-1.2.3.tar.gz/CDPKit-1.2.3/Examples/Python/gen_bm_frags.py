#!/bin/env python

##
# gen_bm_frags.py 
#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
#
# Permission to use, copy, modify, and/or distribute this software for any purpose with
# or without fee is hereby granted.
# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH REGARD
# TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS. IN
# NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL
# DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
# IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
# CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
##


import sys
import argparse

import CDPL.Chem as Chem

    
# performs Bemis-Murcko framework analyis for a given input molecule
def bmAnalyze(molgraph: Chem.MolecularGraph, keep_hs: bool) -> Chem.BemisMurckoAnalyzer:
    Chem.calcBasicProperties(molgraph, False)  # calculate basic molecular properties (if not yet done)
    
    # create an instance of the class implementing the Bemis-Murcko analyis rules
    bm_analyzer = Chem.BemisMurckoAnalyzer()

    # apply -H option
    bm_analyzer.stripHydrogens(not keep_hs)

    # perform analysis
    bm_analyzer.analyze(molgraph)

    return bm_analyzer
    
def getLogMessage(bm_analyzer: Chem.BemisMurckoAnalyzer, mol_id: str) -> str:
    msg =  f'- Molecule {mol_id}\n'
    msg += f'  Frameworks:   {bm_analyzer.getFrameworks().getSize()}\n'
    msg += f'  Ring Systems: {bm_analyzer.getRingSystems().getSize()}\n'
    msg += f'  Linkers:      {bm_analyzer.getLinkers().getSize()}\n'
    msg += f'  Side Chains:  {bm_analyzer.getSideChains().getSize()}'
     
    return msg

def parseArgs() -> argparse.Namespace:
    def strtobool(value: str) -> bool:
        value = value.lower()
        
        if value in ('y', 'yes', 'on', '1', 'true', 't'):
            return True
        
        return False
    
    parser = argparse.ArgumentParser(description='Analyzes the framework of given input molecules according to the rules established by Bemis and Murcko.')

    parser.add_argument('-i',
                        dest='in_file',
                        required=True,
                        metavar='<file>',
                        help='Input molecule file')
    parser.add_argument('-o',
                        dest='out_file',
                        required=True,
                        metavar='<file>',
                        help='Fragment output file')
    parser.add_argument('-v',
                        dest='verbose',
                        required=False,
                        action='store_true',
                        default=False,
                        help='Verbose output (default: false)')
    parser.add_argument('-m',
                        dest='output_mol',
                        required=False,
                        action='store_true',
                        default=False,
                        help='Output the input molecule before the resulting fragments (default: false)')
    parser.add_argument('-H',
                        dest='keep_hs',
                        required=False,
                        action='store_true',
                        default=False,
                        help='Keep hydrogen atoms (default: false)')
    parser.add_argument('-f',
                        dest='output_fws',
                        required=False,
                        metavar='<true|false>',
                        type=lambda x:bool(strtobool(x)),
                        default=True,
                        help='Output molecule frameworks (default: true)')
    parser.add_argument('-r',
                        dest='output_rsys',
                        required=False,
                        action='store_true',
                        default=False,
                        help='Output molecule ring systems (default: false)')
    parser.add_argument('-s',
                        dest='output_schns',
                        required=False,
                        action='store_true',
                        default=False,
                        help='Output molecule side chains (default: false)')
    parser.add_argument('-l',
                        dest='output_lnks',
                        required=False,
                        action='store_true',
                        default=False,
                        help='Output ring system linkers (default: false)')
    parser.add_argument('-x',
                        dest='atom_mpg',
                        required=False,
                        action='store_true',
                        default=False,
                        help='Label atoms with atom mapping IDs (default: false)')
    parser.add_argument('-c',
                        dest='carbonize',
                        required=False,
                        action='store_true',
                        default=False,
                        help='Transform fragments into carbon skeletons (default: false)')
   
    return parser.parse_args()

# converts all atoms into carbons and equalizes other basic atom/bond properties
def carbonize(molgraph: Chem.MolecularGraph) -> None:
    for atom in molgraph.atoms:
        Chem.setSymbol(atom, 'C')
        Chem.setType(atom, Chem.AtomType.C)
        Chem.clearFormalCharge(atom)
        Chem.setImplicitHydrogenCount(atom, 0)
        Chem.setAromaticityFlag(atom, False)
        Chem.setHybridizationState(atom, Chem.HybridizationState.UNKNOWN)
        Chem.clearStereoDescriptor(atom)
        Chem.clear3DCoordinates(atom)
        Chem.clear3DCoordinatesArray(atom)

    for bond in molgraph.bonds:
        Chem.setOrder(bond, 1)
        Chem.setAromaticityFlag(bond, False)
        Chem.clearStereoDescriptor(bond)
        Chem.set2DStereoFlag(bond, Chem.BondStereoFlag.PLAIN)
        
def outputFragments(frags: Chem.FragmentList, writer: Chem.MolecularGraphWriter, mol_name: str, mol_id: str, frag_type: str) -> None:
    for i in range(len(frags)):
        frag = frags[i]

        # set fragment name encoding the parent molecule and fragment number
        Chem.setName(frag, mol_name + f'_{frag_type}#{i + 1}') 
                        
        # calculate (if not present) some basic properties of the output fragments
        # that might be required for writing (output format dependent)
        Chem.perceiveSSSR(frag, False)
        Chem.perceiveComponents(frag, False)

        try:
            # write fragment
            if not writer.write(frag):
                sys.exit('Error: writing fragment of molecule {mol_id} failed')

        except Exception as e: # handle exception raised in case of severe write errors
            sys.exit(f'Error: writing fragment of molecule {mol_id} failed: {str(e)}')  

def main() -> None:
    args = parseArgs() # process command line arguments
        
    # create reader for input molecules (format specified by file extension)
    reader = Chem.MoleculeReader(args.in_file) 

    # create writer for output fragments (format specified by file extension)
    writer = Chem.MolecularGraphWriter(args.out_file) 
  
    if args.atom_mpg: # enable atom mapping IDs for SMILES output
        Chem.setSMILESMolOutputAtomMappingIDParameter(writer, True)

    # write canonical SMILES
    Chem.setSMILESOutputCanonicalFormParameter(writer, True)
    
    # create instances of the default implementation of the Chem.Molecule interface for the input molecule
    in_mol = Chem.BasicMolecule()
    i = 1
    
    try:
        # read and process molecules one after the other until the end of input has been reached (or a severe error occurs)
        while reader.read(in_mol):
            # compose a molecule identifier
            mol_name = Chem.getName(in_mol).strip() 

            if mol_name == '':
                mol_name = f'Mol#{i}'
                mol_id = f'#{i}'  # fallback if name is empty or not available
            else:
                mol_id = f'\'{mol_name}\' (#{i})'
         
            try:
                # perform Bemis-Murcko analysis of in_mol
                bm_analyzer = bmAnalyze(in_mol, args.keep_hs) 

                if args.verbose:
                    print(getLogMessage(bm_analyzer, mol_id))

                # label atoms with atom mapping IDs to see correspondences between fragment and parent molecule atoms
                if args.atom_mpg: 
                    for atom in in_mol.atoms:
                        Chem.setAtomMappingID(atom, atom.getIndex() + 1)

                try:
                    if args.output_mol: # write input molecule before the obtained fragments
                        if not writer.write(in_mol):
                            sys.exit(f'Error: writing molecule {mol_id} failed')

                    if args.carbonize:  # convert all atoms into carbons and equalize other basic atom/bond properties
                        carbonize(in_mol)

                    if args.output_fws:
                        outputFragments(bm_analyzer.getFrameworks(), writer, mol_name, mol_id, 'F')

                    if args.output_rsys:
                        outputFragments(bm_analyzer.getRingSystems(), writer, mol_name, mol_id, 'R')

                    if args.output_lnks:
                        outputFragments(bm_analyzer.getLinkers(), writer, mol_name, mol_id, 'L')
                     
                    if args.output_schns:
                        outputFragments(bm_analyzer.getSideChains(), writer, mol_name, mol_id, 'S')
                             
                except Exception as e: # handle exception raised in case of severe write errors
                    sys.exit(f'Error: writing molecule {mol_id} failed: {str(e)}')
                
            except Exception as e: # handle exception raised in case of severe structure processing errors
                sys.exit(f'Error: processing of molecule {mol_id} failed: {str(e)}')

            i += 1
            
    except Exception as e: # handle exception raised in case of severe read errors
        sys.exit(f'Error: reading of molecule {mol_id} failed: {str(e)}')

    writer.close()
    sys.exit(0)
        
if __name__ == '__main__':
    main()
