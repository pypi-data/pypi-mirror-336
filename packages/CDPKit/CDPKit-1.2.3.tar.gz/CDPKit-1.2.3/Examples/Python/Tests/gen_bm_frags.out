[H:35][C:13](=[C:8]1[C:9]([N:10]([H:11])[c:4]2[c:5]1[c:6]([H:7])[c:1]([H:38])[c:2]([H:37])[c:3]2[H:36])=[O:12])[N:14]([H:34])[c:15]2[c:16]([H:33])[c:17]([H:32])[c:18]([c:19]([H:22])[c:20]2[H:21])[S:23]([N:24]([H:29])[C:25]([H:26])([H:27])[H:28])(=[O:30])=[O:31] ZINC03814443|cdk2_ligands|dock1
[C:8]1([CH2:9][NH:10][c:4]2[c:5]1[cH:6][cH:1][cH:2][cH:3]2)=[CH:13][NH:14][c:15]2[cH:20][cH:19][cH:18][cH:17][cH:16]2 ZINC03814443|cdk2_ligands|dock1_F#1
[CH2:8]1[CH2:9][NH:10][c:4]2[c:5]1[cH:6][cH:1][cH:2][cH:3]2 ZINC03814443|cdk2_ligands|dock1_R#1
[cH:15]1[cH:16][cH:17][cH:18][cH:19][cH:20]1 ZINC03814443|cdk2_ligands|dock1_R#2
[CH3:13][NH2:14] ZINC03814443|cdk2_ligands|dock1_L#1
[H:35][C:13](=[C:8]1[C:9]([N:10]([H:11])[c:4]2[c:5]1[c:6]([H:7])[c:1]([H:38])[c:2]([H:37])[c:3]2[H:36])=[O:12])[N:14]([H:34])[c:15]2[c:16]([H:33])[c:17]([H:32])[c:18]([c:19]([H:22])[c:20]2[H:21])[S:23]([N:24]([H:29])[C:25]([H:26])([H:27])[H:28])(=[O:30])=[O:31] ZINC03814443|cdk2_ligands|dock5
[C:8]1([CH2:9][NH:10][c:4]2[c:5]1[cH:6][cH:1][cH:2][cH:3]2)=[CH:13][NH:14][c:15]2[cH:20][cH:19][cH:18][cH:17][cH:16]2 ZINC03814443|cdk2_ligands|dock5_F#1
[CH2:8]1[CH2:9][NH:10][c:4]2[c:5]1[cH:6][cH:1][cH:2][cH:3]2 ZINC03814443|cdk2_ligands|dock5_R#1
[cH:15]1[cH:16][cH:17][cH:18][cH:19][cH:20]1 ZINC03814443|cdk2_ligands|dock5_R#2
[CH3:13][NH2:14] ZINC03814443|cdk2_ligands|dock5_L#1
[H:35][C:13](=[C:8]1[C:9]([N:10]([H:11])[c:4]2[c:5]1[c:6]([H:7])[c:1]([H:38])[c:2]([H:37])[c:3]2[H:36])=[O:12])[N:14]([H:34])[c:15]2[c:16]([H:33])[c:17]([H:32])[c:18]([c:19]([H:22])[c:20]2[H:21])[S:23]([N:24]([H:29])[C:25]([H:26])([H:27])[H:28])(=[O:30])=[O:31] ZINC03814443|cdk2_ligands|dock9
[C:8]1([CH2:9][NH:10][c:4]2[c:5]1[cH:6][cH:1][cH:2][cH:3]2)=[CH:13][NH:14][c:15]2[cH:20][cH:19][cH:18][cH:17][cH:16]2 ZINC03814443|cdk2_ligands|dock9_F#1
[CH2:8]1[CH2:9][NH:10][c:4]2[c:5]1[cH:6][cH:1][cH:2][cH:3]2 ZINC03814443|cdk2_ligands|dock9_R#1
[cH:15]1[cH:16][cH:17][cH:18][cH:19][cH:20]1 ZINC03814443|cdk2_ligands|dock9_R#2
[CH3:13][NH2:14] ZINC03814443|cdk2_ligands|dock9_L#1
[H:38][C:13](=[C:8]1[C:9]([N:10]([H:11])[c:4]2[c:5]1[c:6]([H:7])[c:1]([H:41])[c:2]([H:40])[c:3]2[H:39])=[O:12])[N:14]([H:37])[c:15]2[c:16]([H:36])[c:17]([H:35])[c:18]([c:19]([H:22])[c:20]2[H:21])[S:23]([N:24]([H:32])[c:25]2[n:26][c:27]([H:31])[c:28]([H:30])[s:29]2)(=[O:34])=[O:33] ZINC03814447|cdk2_ligands|dock1
[C:8]1([CH2:9][NH:10][c:4]2[c:5]1[cH:6][cH:1][cH:2][cH:3]2)=[CH:13][NH:14][c:15]2[cH:20][cH:19][c:18]([cH:17][cH:16]2)[S:23][NH:24][c:25]2[n:26][cH:27][cH:28][s:29]2 ZINC03814447|cdk2_ligands|dock1_F#1
[CH2:8]1[CH2:9][NH:10][c:4]2[c:5]1[cH:6][cH:1][cH:2][cH:3]2 ZINC03814447|cdk2_ligands|dock1_R#1
[cH:15]1[cH:16][cH:17][cH:18][cH:19][cH:20]1 ZINC03814447|cdk2_ligands|dock1_R#2
[cH:27]1[cH:28][s:29][cH:25][n:26]1 ZINC03814447|cdk2_ligands|dock1_R#3
[CH3:13][NH2:14] ZINC03814447|cdk2_ligands|dock1_L#1
[NH2:24][SH:23] ZINC03814447|cdk2_ligands|dock1_L#2
[H:38][C:13](=[C:8]1[C:9]([N:10]([H:11])[c:4]2[c:5]1[c:6]([H:7])[c:1]([H:41])[c:2]([H:40])[c:3]2[H:39])=[O:12])[N:14]([H:37])[c:15]2[c:16]([H:36])[c:17]([H:35])[c:18]([c:19]([H:22])[c:20]2[H:21])[S:23]([N:24]([H:32])[c:25]2[n:26][c:27]([H:31])[c:28]([H:30])[s:29]2)(=[O:34])=[O:33] ZINC03814447|cdk2_ligands|dock4
[C:8]1([CH2:9][NH:10][c:4]2[c:5]1[cH:6][cH:1][cH:2][cH:3]2)=[CH:13][NH:14][c:15]2[cH:20][cH:19][c:18]([cH:17][cH:16]2)[S:23][NH:24][c:25]2[n:26][cH:27][cH:28][s:29]2 ZINC03814447|cdk2_ligands|dock4_F#1
[CH2:8]1[CH2:9][NH:10][c:4]2[c:5]1[cH:6][cH:1][cH:2][cH:3]2 ZINC03814447|cdk2_ligands|dock4_R#1
[cH:15]1[cH:16][cH:17][cH:18][cH:19][cH:20]1 ZINC03814447|cdk2_ligands|dock4_R#2
[cH:27]1[cH:28][s:29][cH:25][n:26]1 ZINC03814447|cdk2_ligands|dock4_R#3
[CH3:13][NH2:14] ZINC03814447|cdk2_ligands|dock4_L#1
[NH2:24][SH:23] ZINC03814447|cdk2_ligands|dock4_L#2
[H:38][C:13](=[C:8]1[C:9]([N:10]([H:11])[c:4]2[c:5]1[c:6]([H:7])[c:1]([H:41])[c:2]([H:40])[c:3]2[H:39])=[O:12])[N:14]([H:37])[c:15]2[c:16]([H:36])[c:17]([H:35])[c:18]([c:19]([H:22])[c:20]2[H:21])[S:23]([N:24]([H:32])[c:25]2[n:26][c:27]([H:31])[c:28]([H:30])[s:29]2)(=[O:34])=[O:33] ZINC03814447|cdk2_ligands|dock2
[C:8]1([CH2:9][NH:10][c:4]2[c:5]1[cH:6][cH:1][cH:2][cH:3]2)=[CH:13][NH:14][c:15]2[cH:20][cH:19][c:18]([cH:17][cH:16]2)[S:23][NH:24][c:25]2[n:26][cH:27][cH:28][s:29]2 ZINC03814447|cdk2_ligands|dock2_F#1
[CH2:8]1[CH2:9][NH:10][c:4]2[c:5]1[cH:6][cH:1][cH:2][cH:3]2 ZINC03814447|cdk2_ligands|dock2_R#1
[cH:15]1[cH:16][cH:17][cH:18][cH:19][cH:20]1 ZINC03814447|cdk2_ligands|dock2_R#2
[cH:27]1[cH:28][s:29][cH:25][n:26]1 ZINC03814447|cdk2_ligands|dock2_R#3
[CH3:13][NH2:14] ZINC03814447|cdk2_ligands|dock2_L#1
[NH2:24][SH:23] ZINC03814447|cdk2_ligands|dock2_L#2
[H:27][c:12]1[c:11]2[C:10](=[C:9]3[C:8]([c:5]4[c:4]([c:3]([H:31])[c:2]([H:32])[c:1]([H:33])[c:6]4[H:7])[N:28]3[H:29])=[O:30])[C:18]([N:17]([H:20])[c:16]2[c:15]([H:21])[c:14]([H:22])[c:13]1[S:23]([O-:26])(=[O:25])=[O:24])=[O:19] ZINC03814440|cdk2_ligands|dock3
[C:10]1(=[C:9]2[CH2:8][c:5]3[cH:6][cH:1][cH:2][cH:3][c:4]3[NH:28]2)[CH2:18][NH:17][c:16]2[c:11]1[cH:12][cH:13][cH:14][cH:15]2 ZINC03814440|cdk2_ligands|dock3_F#1
[C:10]1(=[C:9]2[CH2:8][c:5]3[cH:6][cH:1][cH:2][cH:3][c:4]3[NH:28]2)[CH2:18][NH:17][c:16]2[c:11]1[cH:12][cH:13][cH:14][cH:15]2 ZINC03814440|cdk2_ligands|dock3_R#1
[H:27][c:12]1[c:11]2[C:10](=[C:9]3[C:8]([c:5]4[c:4]([c:3]([H:31])[c:2]([H:32])[c:1]([H:33])[c:6]4[H:7])[N:28]3[H:29])=[O:30])[C:18]([N:17]([H:20])[c:16]2[c:15]([H:21])[c:14]([H:22])[c:13]1[S:23]([O-:26])(=[O:25])=[O:24])=[O:19] ZINC03814440|cdk2_ligands|dock1
[C:10]1(=[C:9]2[CH2:8][c:5]3[cH:6][cH:1][cH:2][cH:3][c:4]3[NH:28]2)[CH2:18][NH:17][c:16]2[c:11]1[cH:12][cH:13][cH:14][cH:15]2 ZINC03814440|cdk2_ligands|dock1_F#1
[C:10]1(=[C:9]2[CH2:8][c:5]3[cH:6][cH:1][cH:2][cH:3][c:4]3[NH:28]2)[CH2:18][NH:17][c:16]2[c:11]1[cH:12][cH:13][cH:14][cH:15]2 ZINC03814440|cdk2_ligands|dock1_R#1
[H:27][c:12]1[c:11]2[C:10](=[C:9]3[C:8]([c:5]4[c:4]([c:3]([H:31])[c:2]([H:32])[c:1]([H:33])[c:6]4[H:7])[N:28]3[H:29])=[O:30])[C:18]([N:17]([H:20])[c:16]2[c:15]([H:21])[c:14]([H:22])[c:13]1[S:23]([O-:26])(=[O:25])=[O:24])=[O:19] ZINC03814440|cdk2_ligands|dock2
[C:10]1(=[C:9]2[CH2:8][c:5]3[cH:6][cH:1][cH:2][cH:3][c:4]3[NH:28]2)[CH2:18][NH:17][c:16]2[c:11]1[cH:12][cH:13][cH:14][cH:15]2 ZINC03814440|cdk2_ligands|dock2_F#1
[C:10]1(=[C:9]2[CH2:8][c:5]3[cH:6][cH:1][cH:2][cH:3][c:4]3[NH:28]2)[CH2:18][NH:17][c:16]2[c:11]1[cH:12][cH:13][cH:14][cH:15]2 ZINC03814440|cdk2_ligands|dock2_R#1
[H:31][c:4]1[c:5]2[C:9]([C:8]([N:7]([H:30])[c:6]2[c:1]([H:34])[c:2]([H:33])[c:3]1[Br:32])=[O:29])=[N:10][N:11]([H:28])[c:12]1[c:17]([H:18])[c:16]([H:19])[c:15]([c:14]([H:26])[c:13]1[H:27])[S:20]([N:21]([H:23])[H:22])(=[O:24])=[O:25] ZINC03814441|cdk2_ligands|dock5
[C:9]1([CH2:8][NH:7][c:6]2[c:5]1[cH:4][cH:3][cH:2][cH:1]2)=[N:10][NH:11][c:12]2[cH:13][cH:14][cH:15][cH:16][cH:17]2 ZINC03814441|cdk2_ligands|dock5_F#1
[CH2:9]1[CH2:8][NH:7][c:6]2[c:5]1[cH:4][cH:3][cH:2][cH:1]2 ZINC03814441|cdk2_ligands|dock5_R#1
[cH:12]1[cH:13][cH:14][cH:15][cH:16][cH:17]1 ZINC03814441|cdk2_ligands|dock5_R#2
[NH2:10][NH2:11] ZINC03814441|cdk2_ligands|dock5_L#1
[H:31][c:4]1[c:5]2[C:9]([C:8]([N:7]([H:30])[c:6]2[c:1]([H:34])[c:2]([H:33])[c:3]1[Br:32])=[O:29])=[N:10][N:11]([H:28])[c:12]1[c:17]([H:18])[c:16]([H:19])[c:15]([c:14]([H:26])[c:13]1[H:27])[S:20]([N:21]([H:23])[H:22])(=[O:24])=[O:25] ZINC03814441|cdk2_ligands|dock2
[C:9]1([CH2:8][NH:7][c:6]2[c:5]1[cH:4][cH:3][cH:2][cH:1]2)=[N:10][NH:11][c:12]2[cH:13][cH:14][cH:15][cH:16][cH:17]2 ZINC03814441|cdk2_ligands|dock2_F#1
[CH2:9]1[CH2:8][NH:7][c:6]2[c:5]1[cH:4][cH:3][cH:2][cH:1]2 ZINC03814441|cdk2_ligands|dock2_R#1
[cH:12]1[cH:13][cH:14][cH:15][cH:16][cH:17]1 ZINC03814441|cdk2_ligands|dock2_R#2
[NH2:10][NH2:11] ZINC03814441|cdk2_ligands|dock2_L#1
[H:31][c:4]1[c:5]2[C:9]([C:8]([N:7]([H:30])[c:6]2[c:1]([H:34])[c:2]([H:33])[c:3]1[Br:32])=[O:29])=[N:10][N:11]([H:28])[c:12]1[c:17]([H:18])[c:16]([H:19])[c:15]([c:14]([H:26])[c:13]1[H:27])[S:20]([N:21]([H:23])[H:22])(=[O:24])=[O:25] ZINC03814441|cdk2_ligands|dock4
[C:9]1([CH2:8][NH:7][c:6]2[c:5]1[cH:4][cH:3][cH:2][cH:1]2)=[N:10][NH:11][c:12]2[cH:13][cH:14][cH:15][cH:16][cH:17]2 ZINC03814441|cdk2_ligands|dock4_F#1
[CH2:9]1[CH2:8][NH:7][c:6]2[c:5]1[cH:4][cH:3][cH:2][cH:1]2 ZINC03814441|cdk2_ligands|dock4_R#1
[cH:12]1[cH:13][cH:14][cH:15][cH:16][cH:17]1 ZINC03814441|cdk2_ligands|dock4_R#2
[NH2:10][NH2:11] ZINC03814441|cdk2_ligands|dock4_L#1
[H:34][c:27]1[c:28]2[c:29]([c:24]3[C:23]([C:37]([N:36]([H:39])[c:25]3[c:26]1[H:35])=[O:38])=[N:22][N:21]([H:40])[c:3]1[c:4]([H:20])[c:5]([H:19])[c:6]([c:1]([H:42])[c:2]1[H:41])[C:7]([H:18])([H:17])[S:8]([N:9]([H:14])[C:10]([H:13])([H:12])[H:11])(=[O:15])=[O:16])[s:30][c:31]([H:33])[n:32]2 ZINC03814444|cdk2_ligands|dock6
[C:23]1([CH2:37][NH:36][c:25]2[c:24]1[c:29]1[c:28]([cH:27][cH:26]2)[n:32][cH:31][s:30]1)=[N:22][NH:21][c:3]1[cH:4][cH:5][cH:6][cH:1][cH:2]1 ZINC03814444|cdk2_ligands|dock6_F#1
[cH:1]1[cH:2][cH:3][cH:4][cH:5][cH:6]1 ZINC03814444|cdk2_ligands|dock6_R#1
[CH2:23]1[CH2:37][NH:36][c:25]2[c:24]1[c:29]1[c:28]([cH:27][cH:26]2)[n:32][cH:31][s:30]1 ZINC03814444|cdk2_ligands|dock6_R#2
[NH2:21][NH2:22] ZINC03814444|cdk2_ligands|dock6_L#1
[H:34][c:27]1[c:28]2[c:29]([c:24]3[C:23]([C:37]([N:36]([H:39])[c:25]3[c:26]1[H:35])=[O:38])=[N:22][N:21]([H:40])[c:3]1[c:4]([H:20])[c:5]([H:19])[c:6]([c:1]([H:42])[c:2]1[H:41])[C:7]([H:18])([H:17])[S:8]([N:9]([H:14])[C:10]([H:13])([H:12])[H:11])(=[O:15])=[O:16])[s:30][c:31]([H:33])[n:32]2 ZINC03814444|cdk2_ligands|dock1
[C:23]1([CH2:37][NH:36][c:25]2[c:24]1[c:29]1[c:28]([cH:27][cH:26]2)[n:32][cH:31][s:30]1)=[N:22][NH:21][c:3]1[cH:4][cH:5][cH:6][cH:1][cH:2]1 ZINC03814444|cdk2_ligands|dock1_F#1
[cH:1]1[cH:2][cH:3][cH:4][cH:5][cH:6]1 ZINC03814444|cdk2_ligands|dock1_R#1
[CH2:23]1[CH2:37][NH:36][c:25]2[c:24]1[c:29]1[c:28]([cH:27][cH:26]2)[n:32][cH:31][s:30]1 ZINC03814444|cdk2_ligands|dock1_R#2
[NH2:21][NH2:22] ZINC03814444|cdk2_ligands|dock1_L#1
[H:34][c:27]1[c:28]2[c:29]([c:24]3[C:23]([C:37]([N:36]([H:39])[c:25]3[c:26]1[H:35])=[O:38])=[N:22][N:21]([H:40])[c:3]1[c:4]([H:20])[c:5]([H:19])[c:6]([c:1]([H:42])[c:2]1[H:41])[C:7]([H:18])([H:17])[S:8]([N:9]([H:14])[C:10]([H:13])([H:12])[H:11])(=[O:15])=[O:16])[s:30][c:31]([H:33])[n:32]2 ZINC03814444|cdk2_ligands|dock10
[C:23]1([CH2:37][NH:36][c:25]2[c:24]1[c:29]1[c:28]([cH:27][cH:26]2)[n:32][cH:31][s:30]1)=[N:22][NH:21][c:3]1[cH:4][cH:5][cH:6][cH:1][cH:2]1 ZINC03814444|cdk2_ligands|dock10_F#1
[cH:1]1[cH:2][cH:3][cH:4][cH:5][cH:6]1 ZINC03814444|cdk2_ligands|dock10_R#1
[CH2:23]1[CH2:37][NH:36][c:25]2[c:24]1[c:29]1[c:28]([cH:27][cH:26]2)[n:32][cH:31][s:30]1 ZINC03814444|cdk2_ligands|dock10_R#2
[NH2:21][NH2:22] ZINC03814444|cdk2_ligands|dock10_L#1
[H:30][C:17]12[c:16]3[c:15]([C:28]([N:18]1[C:19]([H:27])([H:26])[C:20]([H:25])([H:24])[C:21]2([H:23])[H:22])=[O:29])[c:14]([H:31])[c:13]([H:32])[c:12]([H:33])[c:11]3[N:10]([H:34])[C:9]([N:8]([H:36])[c:5]3[c:6]([H:7])[c:1]([H:39])[c:2]([H:38])[c:3]([H:37])[n:4]3)=[O:35] ZINC03814449|cdk2_ligands|dock2
[CH2:21]1[CH2:20][CH2:19][N:18]2[CH:17]1[c:16]1[c:15]([CH2:28]2)[cH:14][cH:13][cH:12][c:11]1[NH:10][CH2:9][NH:8][c:5]1[cH:6][cH:1][cH:2][cH:3][n:4]1 ZINC03814449|cdk2_ligands|dock2_F#1
[cH:1]1[cH:2][cH:3][n:4][cH:5][cH:6]1 ZINC03814449|cdk2_ligands|dock2_R#1
[CH2:21]1[CH2:20][CH2:19][N:18]2[CH:17]1[c:16]1[c:15]([CH2:28]2)[cH:14][cH:13][cH:12][cH:11]1 ZINC03814449|cdk2_ligands|dock2_R#2
[CH2:9]([NH2:8])[NH2:10] ZINC03814449|cdk2_ligands|dock2_L#1
[H:30][C:17]12[c:16]3[c:15]([C:28]([N:18]1[C:19]([H:27])([H:26])[C:20]([H:25])([H:24])[C:21]2([H:23])[H:22])=[O:29])[c:14]([H:31])[c:13]([H:32])[c:12]([H:33])[c:11]3[N:10]([H:34])[C:9]([N:8]([H:36])[c:5]3[c:6]([H:7])[c:1]([H:39])[c:2]([H:38])[c:3]([H:37])[n:4]3)=[O:35] ZINC03814449|cdk2_ligands|dock5
[CH2:21]1[CH2:20][CH2:19][N:18]2[CH:17]1[c:16]1[c:15]([CH2:28]2)[cH:14][cH:13][cH:12][c:11]1[NH:10][CH2:9][NH:8][c:5]1[cH:6][cH:1][cH:2][cH:3][n:4]1 ZINC03814449|cdk2_ligands|dock5_F#1
[cH:1]1[cH:2][cH:3][n:4][cH:5][cH:6]1 ZINC03814449|cdk2_ligands|dock5_R#1
[CH2:21]1[CH2:20][CH2:19][N:18]2[CH:17]1[c:16]1[c:15]([CH2:28]2)[cH:14][cH:13][cH:12][cH:11]1 ZINC03814449|cdk2_ligands|dock5_R#2
[CH2:9]([NH2:8])[NH2:10] ZINC03814449|cdk2_ligands|dock5_L#1
[H:30][C:17]12[c:16]3[c:15]([C:28]([N:18]1[C:19]([H:27])([H:26])[C:20]([H:25])([H:24])[C:21]2([H:23])[H:22])=[O:29])[c:14]([H:31])[c:13]([H:32])[c:12]([H:33])[c:11]3[N:10]([H:34])[C:9]([N:8]([H:36])[c:5]3[c:6]([H:7])[c:1]([H:39])[c:2]([H:38])[c:3]([H:37])[n:4]3)=[O:35] ZINC03814449|cdk2_ligands|dock7
[CH2:21]1[CH2:20][CH2:19][N:18]2[CH:17]1[c:16]1[c:15]([CH2:28]2)[cH:14][cH:13][cH:12][c:11]1[NH:10][CH2:9][NH:8][c:5]1[cH:6][cH:1][cH:2][cH:3][n:4]1 ZINC03814449|cdk2_ligands|dock7_F#1
[cH:1]1[cH:2][cH:3][n:4][cH:5][cH:6]1 ZINC03814449|cdk2_ligands|dock7_R#1
[CH2:21]1[CH2:20][CH2:19][N:18]2[CH:17]1[c:16]1[c:15]([CH2:28]2)[cH:14][cH:13][cH:12][cH:11]1 ZINC03814449|cdk2_ligands|dock7_R#2
[CH2:9]([NH2:8])[NH2:10] ZINC03814449|cdk2_ligands|dock7_L#1